src/components/ PageEvent/PageEvent.tsx

```diff
@@ -1,21 +1,27 @@
-import { PropsWithChildren, useState } from 'react';
+import { useState } from 'react';
 import Clock from '../Clock/Clock';
-import styles from './App.module.scss';
-import { parseUrl, getCurrentTimezone } from '../../utils';
+import styles from './PageEvent.module.scss';
+import { getCurrentTimezone } from '../../utils';
 import Modal from '../Modal/Modal';
 import Timeinfo from '../Timeinfo/Timeinfo';
 import Setevent from '../Setevent/Setevent';
 import FuzzySearch from '../FuzzySearch/FuzzySearch';
 import { timezones } from '../FuzzySearch/timezones';
+import { Moment } from 'moment';
 
 interface PageEventProps {
-  eventUrl: string; // Свойство для URL события
+  date: Moment;
+  timezone: string;
+  event: any;
 }
 
-function PageEvent(props: PropsWithChildren<PageEventProps>) {
-  const { date, timezone } = parseUrl(window.location.href);
-  const [selectedTimezone, setSelectedTimezone] = useState<string>('');
-  const [savedTimezone, setSavedTimezone] = useState<string>(timezone);
+function PageEvent({ date, timezone, event }: PageEventProps) {
+  const decodedEventName = decodeURIComponent(event || '');
```

Comment:
I would expect this to be outside of the component. Also, event can never be undefined, so `|| ''` is redundant